src/python/tests_extended/test_tensor_based.py

```diff
@@ -0,0 +1,1723 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------------------------
+
+import os
+import platform
+import tempfile
+import unittest
+
+import numpy as np
+import pandas as pd
+from nimbusml import Pipeline
+from nimbusml.preprocessing.schema import ColumnSelector
+from nimbusml.preprocessing import ToString, ToKeyImputer, DateTimeSplitter
+from nimbusml.timeseries import TimeSeriesImputer, LagLeadOperator, RollingWindow, ForecastingPivot, ShortDrop
+from nimbusml.preprocessing import (TensorFlowScorer, FromKey, ToKey,
+                                    DateTimeSplitter, OnnxRunner)
+import onnxruntime as rt
+from data_frame_tool import DataFrameTool as DFT
+
+def get_tmp_file(suffix=None):
+    fd, file_name = tempfile.mkstemp(suffix=suffix)
+    fl = os.fdopen(fd, 'w')
+    fl.close()
+    return file_name
+
+class TestAutoMLTransforms(unittest.TestCase):
+    # invalid input test
+    
+    def test_datetimesplitter_bad_input_data(self):
+        df = pd.DataFrame(data=dict(
+            tokens1=[1, 2, 3, 157161600]
+        ))
+
+        cols_to_drop = [
+            'dtHour12', 'dtDayOfWeek', 'dtDayOfQuarter',
+            'dtDayOfYear', 'dtWeekOfMonth', 'dtQuarterOfYear',
+            'dtHalfOfYear', 'dtWeekIso', 'dtYearIso', 'dtMonthLabel',
+            'dtAmPmLabel', 'dtDayOfWeekLabel', 'dtIsPaidTimeOff'
+        ]
+
+        dts = DateTimeSplitter(prefix='dt') << 'tokens1'
+        xf = Pipeline([dts, ColumnSelector(drop_columns=cols_to_drop)])
+        xf.fit(df)
+
+        onnx_path = get_tmp_file('.onnx')
+        onnx_json_path = get_tmp_file('.onnx.json')
+        xf.export_to_onnx(onnx_path,
+                            'com.microsoft.ml',
+                            dst_json=onnx_json_path,
+                            onnx_version='Stable')
+        sess = rt.InferenceSession(onnx_path)
+
+        with self.assertRaisesRegex(Exception, "ONNXRuntimeError"):
+            data1 = np.array([1, 2, 3, -3]).reshape(4,1)
+            pred = sess.run(None, {"tokens1":data1})
+            
+    def test_datetimesplitter_bad_input_type(self):
```

Comment:
>test_datetimesplitter_bad_input_type [](start = 8, length = 36)

Same here, refactor so the whole test is very simple like: df = pd.DataFrame(..my test data..); test_datetimesplitter(df).